pyfar/dsp/dsp.py

```diff
@@ -1386,6 +1386,232 @@ def time_shift(signal, shift, unit='samples'):
     return shifted.reshape(signal.cshape)
 
 
+def find_impulse_response_delay(ir, N=1):
+    """Find the delay in samples of an impulse response.
+    The method relies on the analytic part of the cross-correlation function
+    of the impulse response and it's minimum-phase equivalent, which is zero
+    for the maximum of the correlation function. For sub-sample root finding,
+    the analytic signal is approximated using a polynomial of order ``N``.
+    The algorithm is based on [#]_ with the following modifications:
+
+    1.  Values with negative gradient used for polynolmial fitting are
+        rejected, allowing to use larger part of the signal for fitting.
+    2.  By default a first order polynomial is used, as the slope of the
+        analytic signal should in theory be linear.
+
+
+    Parameters
+    ----------
+    ir : Signal
+        The impulse response
```

Comment:
I'd suggest to use the same name of the parameter as for `find_impulse_response_start`